expansion/src/test/java/no/unit/nva/publication/indexing/AffiliationGenerator.java

```diff
@@ -0,0 +1,68 @@
+package no.unit.nva.publication.indexing;
+
+import static no.unit.nva.expansion.ExpansionConfig.objectMapper;
+import static nva.commons.core.attempt.Try.attempt;
+import static nva.commons.core.ioutils.IoUtils.stringFromResources;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import java.net.URI;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+public class AffiliationGenerator {
+
+    public static final String TOP_LEVEL_SAMPLE_JSON_FILENAME = "framed-json/affiliation_parent_sample.json";
+    public static final String SAMPLE_JSON_FILENAME = "framed-json/affiliation_sample.json";
+    public static final String FIELD_ID = "id";
+    public static final String FIELD_HAS_PART = "/hasPart";
+    public static final String FIELD_FIRST_PART_OF = "/partOf/0";
+
+    public static Map<URI, String> getAffiliationsWithCommonParent(List<URI> childUris, URI parentURI) {
+        Map<URI, String> uriToContent = new java.util.HashMap<>(Collections.emptyMap());
+
+        String parentSample = stringFromResources(Path.of(TOP_LEVEL_SAMPLE_JSON_FILENAME));
+        JsonNode parentJson = attempt(() -> objectMapper.readTree(parentSample)).orElseThrow();
+
+        int i = 0;
+        for (URI childURI : childUris) {
+            JsonNode content = updateFieldsForChilAffiliation(parentURI, childURI);
+
+            uriToContent.put(childURI, content.toString());
+
+            setChildrenForParentAffiliation(parentJson, i, childURI);
```

Comment:
Tip: You can increase "i" inline while accessing it.
`setChildrenForParentAffiliation(parentJson, i,++ childURI);`
and then you dont have to increase it in the next line